services/feeds/queries.ts

```diff
@@ -75,4 +74,44 @@ const useWriteFeed = () => {
   return { postStrategy, loading };
 };
 
-export { useFeeds, useFeed, useWriteFeed };
+const useLikeMutation = () => {
+  const queryClient = useQueryClient();
+
+  return useMutation({
+    mutationFn: likeAStrategy,
+    onMutate: async (strategyId: string) => {
+      const queryKey: any = ['feedDetail', strategyId] as QueryKey;
+
+      await queryClient.cancelQueries(queryKey);
+
+      const previousFeed = queryClient.getQueryData(queryKey);
+
+      queryClient.setQueryData(queryKey, (oldData: any) => {
+        if (!oldData) return oldData;
+        return {
+          ...oldData,
+          isLiked: true,
+          likesCount: oldData.likesCount + 1,
+        };
+      });
+
+      return { previousFeed };
+    },
+    onError: (err, strategyId, context) => {
```

Comment:
Couldn't you pass strategyId to useLikeMutation? And do everything in here instead of doing the mutation in the component